WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingLaunchStoreView.swift

```diff
@@ -73,13 +80,63 @@ struct StoreOnboardingLaunchStoreView: View {
             }
         }
         .navigationTitle(Localization.title)
+        .task {
+            await viewModel.checkEligibilityToPublishStore()
+        }
+    }
+
+    var freeTrialBanner: some View {
+        HStack(alignment: .top, spacing: Layout.FreeTrialBanner.horizontalSpacing) {
+            Image(uiImage: .infoOutlineImage)
+                .resizable()
+                .renderingMode(.template)
+                .foregroundColor(Color(.withColorStudio(.green, shade: .shade80)))
+                .frame(width: Layout.FreeTrialBanner.infoIconSize.width * scale, height: Layout.FreeTrialBanner.infoIconSize.height * scale)
+
+            AttributedText(upgradePlanAttributedString)
+        }
+        .padding(insets: Layout.FreeTrialBanner.padding)
+        .frame(maxWidth: .infinity, alignment: .leading)
+        .background(Color(.bannerBackground))
+        .onTapGesture {
+            viewModel.didTapUpgrade()
+        }
     }
 }
 
 private extension StoreOnboardingLaunchStoreView {
+    var upgradePlanAttributedString: NSAttributedString {
```

Comment:
Nit: Should we move this to a lazy var in the view model instead? We should avoid computed variables on SwiftUI views to optimize the performance for re-rendering.

Reply:
I understand. Done in 42bdc62e269175bc79c8b04b115e45c7d5482e92